### PR TITLE
Remove unused headers

### PR DESCRIPTION
**Pull request by YYF233333**
As #100564 and #100634 are merged smoothly, I think it is time to pace a little bit faster. Here are cleanups for other parts of the engine.

### CODE REVIEW

File: drivers/gles3/storage/config.cpp [hunk 1]
```diff
@@ -28,12 +28,12 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
 /**************************************************************************/
 
+#include "core/config/project_settings.h"
```
Comment (AThousandShips) [inline drivers/gles3/storage/config.cpp:31]: Move down below, it should be after the other headers

File: editor/add_metadata_dialog.cpp [hunk 2]
```diff
@@ -29,6 +29,7 @@
 /**************************************************************************/
 
 #include "add_metadata_dialog.h"
+#include "editor/themes/editor_scale.h"
```
Comment (AThousandShips) [inline editor/add_metadata_dialog.cpp:32]:
```suggestion

#include "editor/themes/editor_scale.h"
```

File: editor/export/lipo.cpp [hunk 3]
```diff
@@ -29,6 +29,7 @@
 /**************************************************************************/
 
 #include "lipo.h"
+#include "macho.h"
```
Comment (AThousandShips) [inline editor/export/lipo.cpp:32]:
```suggestion

#include "macho.h"
```

File: scene/3d/fog_volume.cpp [hunk 4]
```diff
@@ -29,6 +29,7 @@
 /**************************************************************************/
 
 #include "fog_volume.h"
+#include "scene/main/viewport.h"
```
Comment (AThousandShips) [inline scene/3d/fog_volume.cpp:32]:
```suggestion

#include "scene/main/viewport.h"
```

File: scene/3d/physics/physical_bone_3d.cpp [hunk 5]
```diff
@@ -29,6 +29,7 @@
 /**************************************************************************/
 
 #include "physical_bone_3d.h"
+#include "scene/3d/physical_bone_simulator_3d.h"
```
Comment (AThousandShips) [inline scene/3d/physics/physical_bone_3d.cpp:32]:
```suggestion

#include "scene/3d/physical_bone_simulator_3d.h"
```

File: scene/main/http_request.cpp [hunk 6]
```diff
@@ -29,7 +29,6 @@
 /**************************************************************************/
 
 #include "http_request.h"
```
Comment (AThousandShips) [inline scene/main/http_request.cpp:31]:
```suggestion
#include "http_request.h"

```

File: scene/resources/shader_include.cpp [hunk 7]
```diff
@@ -29,7 +29,6 @@
 /**************************************************************************/
 
 #include "shader_include.h"
```
Comment (AThousandShips) [inline scene/resources/shader_include.cpp:31]:
```suggestion
#include "shader_include.h"

```

**Review by AThousandShips — DISMISSED** [review]

File: drivers/gles3/storage/config.cpp (hunk 1 above)
Comment (YYF233333) [reply drivers/gles3/storage/config.cpp]: This head is required by `rasterizer_gles3.h` but not directly, should be some complex dependencies, I revert this change and leave it for following works.

Comment (AThousandShips) [reply drivers/gles3/storage/config.cpp]: Then that's an issue to solve separately I would say

**Review by Repiteo — COMMENTED** [review]
Couldn't select them all, but quite a few instances of `<cstdint>`/`<stdint.h>`/`<stddef.h>` are redundant when `core/typedefs.h` is already included. Feel free to disregard if that feels outside the scope of this PR

**Comment by YYF233333** [comment]
> Couldn't select them all, but quite a few instances of `<cstdint>`/`<stdint.h>`/`<stddef.h>` are redundant when `core/typedefs.h` is already included. Feel free to disregard if that feels outside the scope of this PR

Yes, I think there should be some more redundant headers, but as they are actually used clangd would not pop a warning so it cost some more to distinguish them. I plan to go through all headers a second time for foward declarations, will try catching missing ones.

**Review by Repiteo — APPROVED** [review]
436 files changed? YOLO

**Review by AThousandShips — APPROVED** [review]
LGTM

**Comment by Repiteo** [comment]
In the meantime, be sure to squash those later commits. The first 3 being separate because of their scope is fine, just integrate the latter 3 into them as necessary

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**